lib/marc/xml_parsers.rb

```diff
@@ -46,18 +52,28 @@ def start_element_namespace name, attributes = [], prefix = nil, uri = nil, ns =
          when 'leader' then @current_element = :leader
          when 'controlfield'
            @current_element=:field
-           @record[:field] = MARC::ControlField.new(attributes["tag"])
+           @record[:field] = new_field(MARC::ControlField, attributes["tag"])
+
          when 'datafield'
-           @record[:field] = MARC::DataField.new(attributes["tag"], attributes['ind1'], attributes['ind2'])
+           @record[:field] = new_field(MARC::DataField, attributes["tag"], attributes['ind1'], attributes['ind2'])
          when 'subfield'
            @current_element=:subfield
-           @record[:subfield] = MARC::Subfield.new(attributes['code'])
+           @record[:subfield] = new_field(MARC::Subfield, attributes['code'])
          end
        end
-     end
+    end
 
+    def new_field(f, *args)
+      begin
+        f.new(*args)
+      rescue MARC::Exception => e
+        @record[:error] = e.message
+        nil
```

Comment:
What is @record there? I'm getting confused!  About whether this is actually the right place to store presence of an error and info about it. 

If we're going to store the message, we probably ought to store the entire  exception `e`, so you have the backtrace and exception class too, along with any other exception-class-specific fields.

Reply:
We need we somewhere to hold the state of errors of a record build.  Since `@record` is already defined via the init function, and it's already being used to hold state  associated to record build, I thought it natural to use it.  When I read the code in context it seems to make sense to me.  Would you prefer it there be an @errors instance variable instead?

I'm thinking it could also be a an array instead of a single error as I have it now, but I did not want to make it too complicated, that that would be more honest/correct.  Storing the entire exception if we are going to do something with it.  But, since I was only interested in printing the message, I didn